From 82f8ef0fa39d2cf34e482471a889443a7be4ac69 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Wed, 14 Dec 2011 20:27:46 +0800
Subject: [PATCH 183/280] ENGR00170228 vpu: Add f/w source code version info
 report for mx6

The major f/w version is increased only when formal f/w is released
from third party on mx6, so we also need to report detail code version.

There is minor vpu API change with the patch.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/Makefile  | 2 +-
 vpu/vpu_lib.c | 6 ++++--
 vpu/vpu_lib.h | 3 ++-
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/vpu/Makefile b/vpu/Makefile
index 2a2f134..7780379 100644
--- a/vpu/Makefile
+++ b/vpu/Makefile
@@ -7,7 +7,7 @@ INCLUDE_LIST:= IMX27ADS IMX51 IMX53 IMX6Q
 OBJ = vpu_io.o vpu_util.o vpu_lib.o vpu_gdi.o
 
 LIBNAME = libvpu
-SONAMEVERSION=2
+SONAMEVERSION=3
 
 ifeq ($(PLATFORM), $(findstring $(PLATFORM), $(INCLUDE_LIST)))
 
diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index f20deb9..73147ae 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -424,7 +424,7 @@ RetCode vpu_SWReset(DecHandle handle, int index)
  */
 RetCode vpu_GetVersionInfo(vpu_versioninfo * verinfo)
 {
-	Uint32 ver;
+	Uint32 ver, fw_code = 0;
 	Uint16 pn, version;
 	RetCode ret = RETCODE_SUCCESS;
 	char productstr[18] = { 0 };
@@ -445,6 +445,8 @@ RetCode vpu_GetVersionInfo(vpu_versioninfo * verinfo)
 	while (VpuReadReg(BIT_BUSY_FLAG)) ;
 
 	ver = VpuReadReg(RET_VER_NUM);
+	if (cpu_is_mx6q())
+		fw_code = VpuReadReg(RET_FW_CODE_REV);
 	UnlockVpu(vpu_semap);
 
 	if (ver == 0)
@@ -477,7 +479,7 @@ RetCode vpu_GetVersionInfo(vpu_versioninfo * verinfo)
 		verinfo->fw_major = (version >> 12) & 0x0f;
 		verinfo->fw_minor = (version >> 8) & 0x0f;
 		verinfo->fw_release = version & 0xff;
-
+		verinfo->fw_code = fw_code;
 		verinfo->lib_major = (VPU_LIB_VERSION_CODE >> (12)) & 0x0f;
 		verinfo->lib_minor = (VPU_LIB_VERSION_CODE >> (8)) & 0x0f;
 		verinfo->lib_release = (VPU_LIB_VERSION_CODE) & 0xff;
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index 074ae9c..094bcd3 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -626,6 +626,7 @@ typedef struct vpu_versioninfo {
 	int fw_major;		/* firmware major version */
 	int fw_minor;		/* firmware minor version */
 	int fw_release;		/* firmware release version */
+	int fw_code;		/* firmware checkin code number */
 	int lib_major;		/* library major version */
 	int lib_minor;		/* library minor version */
 	int lib_release;	/* library release version */
@@ -649,7 +650,7 @@ typedef struct vpu_versioninfo {
  * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 3, 6)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 3, 7)
 
 extern unsigned int system_rev;
 
-- 
1.8.0

